elastica/contact_forces.py

```diff
@@ -725,3 +861,120 @@ def apply_contact(
+    """
+    This class is for applying contact forces between rod-sphere.
+    First system is always rod and second system is always sphere.
+    In addition to the contact forces, user can define apply friction forces between rod and sphere that
+    are in contact. For details on friction model refer to this [1]_.
+    Notes
+    -----
+    The `velocity_damping_coefficient` is set to a high value (e.g. 1e4) to minimize slip and simulate stiction
+    (static friction), while friction_coefficient corresponds to the Coulombic friction coefficient.
+    Examples
+    --------
+    How to define contact between rod and sphere.
+    >>> simulator.detect_contact_between(rod, sphere).using(
+    ...    RodSphereContact,
+    ...    k=1e4,
+    ...    nu=10,
+    ... )
+    .. [1] Preclik T., Popa Constantin., Rude U., Regularizing a Time-Stepping Method for Rigid Multibody Dynamics, Multibody Dynamics 2011, ECCOMAS. URL: https://www10.cs.fau.de/publications/papers/2011/Preclik_Multibody_Ext_Abstr.pdf
+    """
+
+    def __init__(
+        self,
+        k: float,
+        nu: float,
+        velocity_damping_coefficient=0.0,
+        friction_coefficient=0.0,
+    ):
+        """
+        Parameters
+        ----------
+        k : float
+            Contact spring constant.
+        nu : float
+            Contact damping constant.
+        velocity_damping_coefficient : float
+            Velocity damping coefficient between rigid-body and rod contact is used to apply friction force in the
+            slip direction.
+        friction_coefficient : float
+            For Coulombic friction coefficient for rigid-body and rod contact.
+        """
+        super(RodSphereContact, self).__init__()
+        self.k = k
+        self.nu = nu
+        self.velocity_damping_coefficient = velocity_damping_coefficient
+        self.friction_coefficient = friction_coefficient
+
+    def _check_systems_validity(
+        self,
+        system_one: SystemType,
+        system_two: AllowedContactType,
+    ) -> None:
+        """
+        This checks the contact order and type of a SystemType object and an AllowedContactType object.
+        For the RodSphereContact class first_system should be a rod and second_system should be a sphere.
+        Parameters
+        ----------
+        system_one
+            SystemType
+        system_two
+            AllowedContactType
+        """
+        if not issubclass(system_one.__class__, RodBase) or not issubclass(
+            system_two.__class__, Sphere
+        ):
+            raise TypeError(
+                "Systems provided to the contact class have incorrect order/type. \n"
+                " First system is {0} and second system is {1}. \n"
+                " First system should be a rod, second should be a sphere".format(
+                    system_one.__class__, system_two.__class__
+                )
+            )
+
+    def apply_contact(self, system_one: RodType, system_two: SystemType):
```

Comment:
Some of the other contact methods have `*args` and `*kwargs` . I think you are passing time through them, is this correct? If that is the case why not passing time explicitly. Why do we need `*args` and `*kwargs`. Also sphere contact doesn't have args and kwargs.

Reply:
Removed `*args` and `**kwargs` from `apply_contact()` as they are not necessary

Reply:
We should keep `*args` and `*kwargs` like `forcing`, `damping`, or `constraints`. It is an interface for extra user parameters. I'm unsure if anyone will ever use it, but the feature is there for other modules.

We are not passing time, because we don't have any `ramping` behavior for contact, but it might be useful in the future.

@armantekinalp Any thoughts?